src/main/java/com/glucoseguardian/webbackend/application/adapter/LoginAdapter.java

```diff
@@ -0,0 +1,78 @@
+package com.glucoseguardian.webbackend.application.adapter;
+
+import com.glucoseguardian.webbackend.application.service.LoginService;
+import com.glucoseguardian.webbackend.storage.dto.LoginInputDto;
+import com.glucoseguardian.webbackend.storage.dto.LoginOutputDto;
+import com.glucoseguardian.webbackend.storage.dto.MessaggioDto;
+import com.glucoseguardian.webbackend.storage.dto.TotpDto;
+import com.glucoseguardian.webbackend.storage.entity.Admin;
+import com.glucoseguardian.webbackend.storage.entity.Dottore;
+import com.glucoseguardian.webbackend.storage.entity.Paziente;
+import com.glucoseguardian.webbackend.storage.entity.Tutore;
+
+public class LoginAdapter implements LoginAdapterInterface {
+  private LoginService adaptee;
+
+  @Override
+  public LoginOutputDto login(LoginInputDto input) {
+    String email = input.getEmail();
+    String password = input.getPassword();
+    Integer otp = input.getOtp();
+
+    String idUtente;
+    int tipoUtente;
+
+    Admin adminRes = adaptee.loginAdmin(email, password, otp);
+    if (adminRes != null) {
+      idUtente = adminRes.getCodiceFiscale();
+      tipoUtente = 0;
```

Comment:
I think it's time to have an enum for this, with a toDBInteger function and a fromDBInteger function, it's unclear expressed like this